app-modules/prospect/src/Filament/Resources/ProspectResource/Pages/ViewProspect.php

```diff
@@ -138,7 +138,7 @@ protected function getHeaderActions(): array
         return [
             ConvertToStudent::make()->visible(fn (Prospect $record) => ! $record->student()->exists() && Feature::active('convert_prospect_to_student')),
             DisassociateStudent::make()->visible(fn (Prospect $record) => $record->student()->exists() && Feature::active('convert_prospect_to_student')),
-            EditAction::make(),
+            EditAction::make()->visible(fn (Prospect $record) => ! $record->student()->exists()),
```

Comment:
This actually is probably not needed because the Policy will return false for `update`. Please confirm.